### ISSUE TITLE
labels and decoder_input_ids

### ISSUE THREAD

**Issue by AI678** [huggingface/transformers#7865]
# ❓ Questions & Help

## Details

what is the difference between labels  and decoder_input_ids for EncoderDecoderModel in the text summarization task?

**A link to original question on the forum/Stack Overflow**:

**Comment by marcoabrate** [comment]
I also get confused by the notation and would appreciate some clarifications.
Thank you.

**Comment by marcoabrate** [comment]
@AI678 I think I got it. Do not take my words as absolute truth since I am new to the subject.
If you are talking about a full Transformer architecture (e.g. BART, T5, PEGASUS), the labels are the token ids to which you compare the logits generated by the Decoder in order to compute the cross-entropy loss. This should be the only input necessary during training or fine tuning.
On the other hand, the decoder_input_ids have the exact same structure of the labels, but are shifted one position to the right in order to add the <start of sentence> token. The decoder_input_ids are then passed to the decoder (along with the mask, which masks all unseen tokens!!). The first input of the decoder will be the <start of sentence> token so that it starts generating the sentence.

Hope this help, waiting for somebody to acknowledge my answer.

**Comment by LysandreJik** [comment]
Hi, the glossary has been completed with these terms, you can check it out [here](https://huggingface.co/transformers/glossary.html), let me know if it's still not clear enough.
